### PR TITLE
Python 3.9 support and away with 3.6

### PR DESCRIPTION
**Pull request by hirosassa**
## What is this PR

- away with Python 3.6 and support 3.9
  - gokart supports 3.7 or higher, so it is reasonable, IMO.
- fix GitHub Actions test and introduce tox

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,11 +15,14 @@ style = "pep440"
 pattern = "^(?P<base>\\d+\\.\\d+\\.\\d+)"
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = "^3.7"
```
Comment (hirosassa) [inline pyproject.toml:18]: away with python 3.6

File: pyproject.toml [hunk 2]
```diff
@@ -15,11 +15,14 @@ style = "pep440"
...
 gokart = "^=0.2.4"
 boto3 = "*"
 tqdm = "*"
-moto = "==1.3.6"
+
+[tool.poetry.dev-dependencies]
+tox = "*"
+moto = "*"
```
Comment (hirosassa) [inline pyproject.toml:25]: separate dev dependencies

File: .github/workflows/python_test.yml [hunk 3]
```diff
@@ -11,15 +11,22 @@ jobs:
...
       with:
-        python-version: 3.8
-    - name: UnitTest
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install poetry
-        poetry install
-        poetry run python -m unittest discover test/unittest
```
Comment (hirosassa) [inline .github/workflows/python_test.yml:25]: looks wrong test command (didn't run test in prev version)

File: pyproject.toml [hunk 4]
```diff
@@ -15,11 +15,14 @@ style = "pep440"
...
-python = "^3.6"
+python = "^3.7"
 gokart = "^=0.2.4"
 boto3 = "*"
 tqdm = "*"
-moto = "==1.3.6"
+
+[tool.poetry.dev-dependencies]
+tox = "*"
+mock = "*"
```
Comment (hirosassa) [inline pyproject.toml:25]: separate dev dependencies and in: mock out: moto

**Comment by vaaaaanquish** [comment]
LGTM. Thx!

**Review by vaaaaanquish — APPROVED** [review]

**Pull request merged**